tests/utils.js

```diff
@@ -0,0 +1,39 @@
+const { delay } = require('@semapps/ldp');
+
+const arrayOf = (value) => {
+  // If the field is null-ish, we suppose there are no values.
+  if (!value) {
+    return [];
+  }
+  // Return as is.
+  if (Array.isArray(value)) {
+    return value;
+  }
+  // Single value is made an array.
+  return [value];
+};
+
+/**
+ * Call a callback and expect the result object to have all properties in `fieldNames`.
+ * If not, try again after `ms` until `maxTries` is reached.
+ * If `fieldNames` is `undefined`, the return value of `callback` is expected to not be
+ * `undefined`.
+ *
+ * @type {import("./utilTypes").waitForResource}
+ **/
+const waitForResource = async (ms, fieldNames, maxTries, callback) => {
+  for (let i = 0; i < maxTries; i += 1) {
+    const result = await callback();
+    // If a result (and the expected field, if required) is present, return.
+    if (result !== undefined && arrayOf(fieldNames).every((fieldName) => result[fieldName] !== undefined)) {
+      return result;
+    }
+    await delay(ms);
+  }
+  throw new Error('Waiting for resource failed. No results after ' + maxTries + ' tries');
+};
```

Comment:
The `activitypub.actor` and `activitypub.object` have a similar `awaitCreateComplete` action, but yours is more complete, maybe it should be used by these actions ;-)